### PR TITLE
Add `bullseye` & `bullseye-slim` versions

### LINKED ISSUE #356: Plan to 3-bullseye?
Hello, I met a problem with `ruby:3` (equivalent to `ruby:3.0.2-buster`) and Apple Silicon (M1) Mac environment.

https://docs.docker.com/docker-for-mac/release-notes/#known-issues

According to the known issues of Docker for Mac, the `debian:buster`, ancestor image of `ruby:3.0.2-buster` have a problem with libssl and causes a problem in Apple Silicon Macs.
It is noted that the `debian:bullseye` solves the problem about it.

Docker Hub provides `build-deps:bullseye` (that is base image of `ruby:3.0.2-buster`), so I have experimented to build `ruby:3.0.2-bullseye`.
I have confirmed that the libssl-related problem was solved.

スクリーンショット 2021-08-07 15 53 24

I know that the debian bullseye is not yet released as stable (planned 2021-08-14: https://lists.debian.org/debian-devel-announce/2021/07/msg00003.html), but many people using Apple Silicon devices will have met the problem.

Is there a plan to release ruby images based on `debian:bullseye`? Hopefully, next week? (as soon as bullseye released)

Thanks,

### PR DESCRIPTION
**Pull request by ojab**
[bullseye is out](https://www.debian.org/News/2021/20210814), let's add it.

Closes #356

### CODE REVIEW

**Comment by ojab** [comment]
```
+ echo '62eefa55030788d409003eccd945ccc408f02fe0d71616ed1c1bdcaf7a2f8a54 *autoconf-2.70.patch'
+ sha256sum --check --strict
sha256sum: WARNING: 1 computed checksum did NOT match
```
in `2.6.8-alpine3.14` CI and looks like I can't restart workflow to recheck. Unrelated, sending to review.

**Comment by tianon** [comment]
As in https://github.com/docker-library/python/pull/633, this should also drop all traces of `stretch` (since we build from source, every variant is a non-trivial amount of added load, so we only maintain max two per distro).

**Comment by tianon** [comment]
(You'll also need to update/run `versions.sh`)

**Comment by ojab** [comment]
Done, CI is green except the same `2.6.8-alpine3.14` failure.

File: Dockerfile.template [hunk 1]
```diff
@@ -111,10 +111,7 @@ RUN set -eux; \
 		make \
 		wget \
 		xz-utils \
-{{ if env.variant | endswith("stretch") then "" else ( -}}
-# https://packages.debian.org/sid/libgdbm-compat-dev (needed for "dbm" core module, but only in Buster+)
 		libgdbm-compat-dev \
```
Comment (tianon) [inline Dockerfile.template:114]: This list can be re-alphabetized now :sweat_smile:

File: versions.sh [hunk 2]
```diff
@@ -69,9 +69,7 @@ for version in "${versions[@]}"; do
 			variants: [
 				(
 					"buster",
-					if env.rcVersion == "2.6" then
-						"stretch"
-					else empty end
+                                        "bullseye"
```
Comment (tianon) [inline versions.sh:72]: These should be in the opposite order, and ought to have matching whitespace :sweat_smile:

```suggestion
					"bullseye",
					"buster"
```

**Review by tianon — COMMENTED** [review]
Very close, just a couple minor nits -- happy to take over from here if you'd prefer. :+1:

File: versions.sh (hunk 2 above)
Comment (ojab) [reply versions.sh:72]: oh, whitespaces :facepalm: 
So AFAIU we want bullseye to be the default one?

Comment (tianon) [reply versions.sh]: Yep!

**Review by tianon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**